### PR TITLE
rsyslog: make additional log server protocol configurable

### PR DESCRIPTION
**Pull request by berendt**
Closes osism/issues#646

### CODE REVIEW

File: roles/rsyslog/templates/71-additional-log-server.conf.j2 [hunk 1]
```diff
@@ -1 +1,5 @@
+{% if rsyslog_additional_protocol|default('udp') == 'udp' %}
```
Comment (osfrickler) [inline roles/rsyslog/templates/71-additional-log-server.conf.j2:1]: Add this var to the role doc?

It is also a bit optimistic to treat everything != udp as tcp, but I'm also not sure it is worth an extra check.

File: roles/rsyslog/templates/71-additional-log-server.conf.j2 [hunk 2]
```diff
@@ -1 +1,5 @@
...
 *.* @{{ rsyslog_additional_host }}:{{ rsyslog_additional_port }}
+{% else %}
+*.* @@{{ rsyslog_additional_host }}:{{ rsyslog_additional_port }}
```
Comment (artificial-intelligence) [inline roles/rsyslog/templates/71-additional-log-server.conf.j2:4]: LGTM, but:

according to this [post](https://www.rsyslog.com/sending-messages-to-a-remote-syslog-server/) this is a legacy format that should not be used anymore.
It seems that the new format is more readable. so we might want to adapt this in the future.

**Review by artificial-intelligence — APPROVED** [review]

File: roles/rsyslog/templates/71-additional-log-server.conf.j2 (hunk 2 above)
Comment (berendt) [reply roles/rsyslog/templates/71-additional-log-server.conf.j2:4]: @artificial-intelligence Please open an issue for this task in osism/issues.

Comment (artificial-intelligence) [reply roles/rsyslog/templates/71-additional-log-server.conf.j2:4]: https://github.com/osism/issues/issues/654

File: roles/rsyslog/defaults/main.yml [hunk 3]
```diff
@@ -14,8 +14,15 @@ rsyslog_package_name: rsyslog
...
+rsyslog_fluentd_port: 5140
 
 fluentd_port: "{{ rsyslog_fluentd_port }}"
 fluentd_host: "{{ rsyslog_fluentd_host }}"
+
+##########################
+# additional rsyslog
+
+rsyslog_additional_protocol: udp
+rsyslog_additional_host: ""
```
Comment (osfrickler) [inline roles/rsyslog/defaults/main.yml:27]: We just had a discussion in kolla in a similar situation. I think it is better to leave the variable undefined by default, the code does already check whether it is defined.

Comment (berendt) [reply roles/rsyslog/defaults/main.yml:27]: CI is not happy with undefined variable because of the |length check in the template.

Comment (osfrickler) [reply roles/rsyslog/defaults/main.yml:27]: hmm, o.k., let's leave it like this for now

**Review by osfrickler — APPROVED** [review]

**Pull request merged**